platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinScriptingModelBuildersRegistrationAction.kt

```diff
@@ -16,25 +16,29 @@
 package org.gradle.kotlin.dsl.tooling.builders
 
 import org.gradle.api.internal.project.ProjectInternal
-
 import org.gradle.configuration.project.ProjectConfigureAction
-
 import org.gradle.kotlin.dsl.support.serviceOf
-
+import org.gradle.kotlin.dsl.tooling.builders.internal.DiscoveredKotlinScriptsModelBuilder
 import org.gradle.tooling.model.kotlin.dsl.KotlinDslModelsParameters
 import org.gradle.tooling.provider.model.ToolingModelBuilderRegistry
+import org.gradle.tooling.provider.model.internal.IntermediateToolingModelProvider
 
 
 class KotlinScriptingModelBuildersRegistrationAction : ProjectConfigureAction {
 
     override fun execute(project: ProjectInternal) {
 
         val builders = project.serviceOf<ToolingModelBuilderRegistry>()
-        builders.register(KotlinBuildScriptModelBuilder)
+        val intermediateToolingModelProvider = project.serviceOf<IntermediateToolingModelProvider>()
```

Comment:
🚀 It would be nice if Gradle didn't have to run this action against every project in the build, just a thought for the future.